kafka/consumer/fetcher.py

```diff
@@ -664,19 +665,34 @@ def _create_fetch_requests(self):
                 log.debug("Adding fetch request for partition %s at offset %d",
                           partition, position)
 
-        if self.config['api_version'] >= (0, 10):
+        if self.config['api_version'] >= (0, 10, 1):
+            version = 3
+        elif self.config['api_version'] >= (0, 10):
             version = 2
         elif self.config['api_version'] == (0, 9):
             version = 1
         else:
             version = 0
         requests = {}
         for node_id, partition_data in six.iteritems(fetchable):
-            requests[node_id] = FetchRequest[version](
-                -1, # replica_id
-                self.config['fetch_max_wait_ms'],
-                self.config['fetch_min_bytes'],
-                partition_data.items())
+            if version < 3:
+                requests[node_id] = FetchRequest[version](
+                    -1,  # replica_id
+                    self.config['fetch_max_wait_ms'],
+                    self.config['fetch_min_bytes'],
+                    partition_data.items())
+            else:
+                # As of version == 3 partitions will be returned in order as
+                # they are requested, so to avoid starvation with
+                # `fetch_max_bytes` option we need this shuffle
+                partition_data = list(partition_data.items())
+                random.shuffle(partition_data)
```

Comment:
Good call on 3.6 ordered dicts--makes sense. 

However, if v3 supports returning partitions in the order requested, would it be a good idea to expose this as an optional "feature"? Obviously default to random, but give the option to fetch partitions in the same order they are requested. 

For example if a user eventually wanted to request partitions in a particular order, knowingly risking starvation, but wanting to use it as a poor-man's prioritization queue? 

Just tossing it out there... it might be a bad idea, and also might require a lot of internal changes in `kafka-python`, I haven't spent enough time thinking about it.